File: catalog/admin/item/update.php
<?php
defined('AUTH') or die('Restricted access');

include_once $_SERVER['DOCUMENT_ROOT'].'/components/catalog/admin/classes/AdminCatalogItem.php';
$ITEM = new AdminCatalogItem();

$item['id'] = intval($SITE->d[5]);
$item['name'] = f_input('name');
$item['section_id'] = intval($_POST['section_id']);
$item['ordering'] = intval($_POST['ordering']);
$item['pub'] = isset($_POST['pub']) ? 1 : 0;
$item['text'] = $_POST['editor1'];
$item['images'] = $_POST['images_order'];


if(isset($_POST['cancel']))
	header('location: /admin/com/catalog/section/'.$item['section_id']);

if (!$item['name']) {
	$SITE->content = 
	'<div class="bg_gray">'.
		'<h1>Ошибка!</h1>'.
		'<div class="tc_container">Не заполнено наименование каталога</div>'.
	'</div>';		
} else {
	$ITEM->update($item);
	header('location: /admin/com/catalog/section/'.$item['section_id']);
}


function f_input($name){
	$post_data = isset($_POST[$name]) ? $_POST[$name] : '';
	return trim(htmlspecialchars(strip_tags($post_data)));
}

File: dan/system/catalog/item/update.py
import sys
sys.path.append('system/catalog/classes')
from system.catalog.classes.Item import Item
from system.catalog.classes.Char import Char


def update(SITE):
    print('FUNCTION -> system-> calalog -> item -> update')

    id = SITE.p[3]

    ITEM = Item(SITE)
    CHAR = Char(SITE)
    item = ITEM.getItem(id)  # Получаем текущий элемент
    section_id = item['section_id']

    if 'cancel' in SITE.post:
        return {'redirect': '/system/catalog/section/' + str(section_id)}

    chars = []
    i = 1
    for key, value in SITE.post.items():
        # Формируем список словарей характеристик
        if key.startswith('char_'):
            if key == 'char_id[]':
                char_dict = {}
                char_dict['id'] = value
                char_dict['item_id'] = id

            if key == 'char_name_id[]':
                char_dict['name_id'] = value
                char_dict['ordering'] = i

            if key == 'char_value[]':
                char_dict['value'] = value
                chars.append(char_dict)
                i += 1

    # Перебираем список
    for char in chars:
        if char['id'] == '':
            CHAR.insertValue(char);
        else:
            CHAR.updateValue(char);
    ITEM.update({
        'id': id,
        'name': SITE.post['name'],
        'text': SITE.post['text'],
        'data': SITE.post['data'],
        'status': SITE.post['status'],
        'ordering': SITE.post['ordering']
    })

    return {'redirect': '/system/catalog/section/' + str(section_id)}
